### PR TITLE
Add Server Request Body support

### PR DESCRIPTION
**Pull request by yassernasc**
'Cherry-picked' a commit from https://github.com/holepunchto/bare-http1/pull/11, the first merged will have it 😅.

### CODE REVIEW

**Review by kasperisager — CHANGES_REQUESTED** [review]
Looks good! I think the only thing missing is chunked encoding support, which you should be able to copy verbatim from `ClientConnection`. As mentioned previously, we should make a proper parser abstraction eventually, but let's leave that for later when all the bits are in place.

**Comment by yassernasc** [comment]
The unified parser abstraction will be a great refactoring, `chunked encoding` support added.

File: lib/client-connection.js [hunk 1]
```diff
@@ -178,6 +178,6 @@ module.exports = class HTTPClientConnection {
   }
 
   _ondrain () {
-    if (this.res) this.res._continueWrite()
+    if (this.req) this.req._continueWrite()
```
Comment (kasperisager) [inline lib/client-connection.js:181]: Good catch!

File: test.js [hunk 2]
```diff
@@ -339,13 +366,19 @@ test('server does a big write', async function (t) {
     host: server.address().address,
     port: server.address().port,
     path: '/'
+  }, (req) => {
+    req.write(Buffer.alloc(2 * 1024 * 1024, 'qwer'))
+    setTimeout(() => { req.end(Buffer.alloc(2 * 1024 * 1024, 'asdf')) })
```
Comment (kasperisager) [inline test.js:371]:
```suggestion
    setImmediate(() => { req.end(Buffer.alloc(2 * 1024 * 1024, 'asdf')) })
```

**Review by kasperisager — CHANGES_REQUESTED** [review]

**Review by kasperisager — APPROVED** [review]

**Pull request merged**